src/picongpu/include/particles/gasProfiles/FromHDF5Impl.hpp

```diff
@@ -74,7 +75,12 @@ struct FromHDF5Impl : public T_ParamClass
     {
         using namespace splash;
         DataConnector &dc = Environment<>::get().DataConnector();
-        FieldTmp& fieldTmp = dc.getData<FieldTmp > (FieldTmp::getName(), true);
+
+        PMACC_CASSERT_MSG(
```

Comment:
I like how with this implementation some overhead from the `Particles` class is removed.
Do I understand correctly, though, that every time that I would instatiate a `FieldTmp` object I would have to add this `cassert`, right? I guess this could not just be moved into the constructor?